src/lib.rs

```diff
@@ -22,3 +22,7 @@ pub use decoding::{decode, decode_header, DecodingKey, TokenData};
 pub use encoding::{encode, EncodingKey};
 pub use header::Header;
 pub use validation::{get_current_timestamp, Validation};
+
+pub use ring::digest::SHA256 as DIGEST_SHA256;
```

Comment:
Let's not expose ring at all. Make an enum with hidden internals and let users use that.

Reply:
Done!